src/cascadia/TerminalApp/JsonUtilsNew.h

```diff
@@ -334,6 +363,18 @@ namespace TerminalApp::JsonUtils
     template<typename T, typename Converter>
     bool GetValue(const Json::Value& json, T& target, Converter&& conv)
     {
+        if constexpr (Detail::DeduceOptional<T>::IsOptional)

```

Comment:
wat the heck is this witchcraft

Reply:
`if constexpr` is SO GOOD. It's like, "only compile this block of code if this thing is true"

it's like `std::enable_if` but for an actual individual block of code instead of how enable_if uses the failure of a type to be well-formed to "hide" it from existence